apptools/feedback/feedbackbot/view.py

```diff
@@ -0,0 +1,97 @@
+"""
+This model implements UI classes and logic for a plugin that enables 
+clients to send feedback messages to a developer team's slack channel.
+"""
+
+from traits.api import Property, Instance
+from traitsui.api import (
+        View, Group, Item, Action, 
+        Label, Controller)
+from traitsui.menu import CancelButton 
+from chaco.api import Plot, ArrayPlotData
+from enable.api import ComponentEditor
+
+from .model import FeedbackMessage
+from .utils import bytes_to_matrix
+
+# ----------------------------------------------------------------------------
+# TraitsUI Actions 
+# ----------------------------------------------------------------------------
+
+send_button = Action(name='Send', action='send', 
+    enabled_when='controller._send_enabled')
+
+# ----------------------------------------------------------------------------
+# TraitsUI Views
+# ----------------------------------------------------------------------------
+
+#: Primary view for the feedback message.
+feedback_msg_view = View(
+    Label('Enter feedback here. All fields are mandatory.'),
+    Group(
+        Group(
+            Item('first_name'),
+            Item('last_name'),
+            Item('organization', 
+                 tooltip='Enter the name of your organization.'),
+            Item('description', 
+                 tooltip='Enter feedback.',
+                 height=200,
+                 springy=True)),
+        Group(
+            Item('controller.screenshot_plot',
+                 editor=ComponentEditor(),
+                 show_label=False)),
+        orientation='horizontal'),
+    buttons=[CancelButton, send_button],
+    width=800,
+    resizable=True)
+
+
+# ----------------------------------------------------------------------------
+# TraitsUI Handler
+# ----------------------------------------------------------------------------
+
+class FeedbackController(Controller):
```

Comment:
You should have a `send` method on the `Controller` rather than the `model` as a general rule.  It works, but it is the wrong way to factor your code.

Reply:
I disagree. `send` is a behavior of the model, and it can be used without a GUI.

Reply:
Let's discuss this more if you want. I'm closing this for now.